File: GeoData-UI/src/app/portal/geo-data-manager/models/geo-data.ts
import { FeatureCollection } from 'geojson';

export interface GeoData extends ItemState {
	ID: string;
	UserID: string;
	Description: string;
	DateCreated: Date;
	DateModified: Date;
	Tags: Array<string>;
	Data: FeatureCollection;	
}

export interface ItemState{
	status: itemStatus;
}

export enum itemStatus{
	clean,
	new,
	dirty
}

File: GeoDataModels/Models/GeoData.cs
using System;
using System.Collections.Generic;

namespace GeoDataModels.Models
{
    public class GeoData
    {
        public GeoData()
        {
            Tags = new List<string>();
            Data = new FeatureCollection();
        }

        public string ID { get; set; }
        public string UserID{get;set;}
        public string Description { get; set; }
        public DateTime DateCreated { get; set; }
        public DateTime DateModified { get; set; }
        public List<string> Tags { get; set; }
        public FeatureCollection Data {get;set;}
    }
}